### PR TITLE
[Backport 1.x] Backporting auto release workflow to 1.x

### PR DESCRIPTION
**Pull request by saratvemulapalli**
### Description
Backporting auto release workflow to 1.x
This workflow automates github releases based on tag, see: https://github.com/opensearch-project/OpenSearch/releases/tag/2.5.0

### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [x] Commits are signed per the DCO using --signoff
- [ ] Commit changes are listed out in CHANGELOG.md file (See: [Changelog](../blob/main/CONTRIBUTING.md#changelog))

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by andrross — APPROVED** [review]

**Comment by saratvemulapalli** [comment]
I have no idea why I cannot merge. 
@reta could you ack as well?

**Comment by andrross** [comment]
It seems something is off with the permissions for the 1.x branch and all changes require @reta's approval

**Review by reta — APPROVED** [review]

**Comment by reta** [comment]
> It seems something is off with the permissions for the 1.x branch and all changes require @reta's approval

Hm ... not sure why, but I am happy to approve in any case

**Comment by reta** [comment]
@andrross oh, I know why, the Github reports that CODEOWNERS file [1] contains errors in first entry, and second entry is myself, this is why :(

[1] https://github.com/opensearch-project/OpenSearch/blob/1.x/.github/CODEOWNERS

**Comment by andrross** [comment]
@reta #6202 and #6203 should fix this, I think

**Pull request merged**
